tests/tests/dispatch_workgroups_indirect.rs

```diff
@@ -68,18 +68,46 @@ static RESET_BIND_GROUPS: GpuTestConfiguration = GpuTestConfiguration::new()
     .run_async(|ctx| async move {
         ctx.device.push_error_scope(wgpu::ErrorFilter::Validation);
 
-        let _ = run_test(&ctx, &[0, 0, 0], true).await;
+        let _ = run_test(&ctx, &[0, 0, 0], true,false).await;
 
         let error = pollster::block_on(ctx.device.pop_error_scope());
         assert!(error.map_or(false, |error| {
             format!("{error}").contains("The current set ComputePipeline with '' label expects a BindGroup to be set at index 0")
         }));
     });
 
+/// Make sure that zero sized buffer validation is raised.
+#[gpu_test]
+static ZERO_SIZED_BUFFER: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(
+        TestParameters::default()
+            .features(wgpu::Features::PUSH_CONSTANTS)
+            .downlevel_flags(
+                wgpu::DownlevelFlags::COMPUTE_SHADERS | wgpu::DownlevelFlags::INDIRECT_EXECUTION,
+            )
+            .limits(wgpu::Limits {
+                max_push_constant_size: 4,
+                ..wgpu::Limits::downlevel_defaults()
+            }),
+    )
+    .run_async(|ctx| async move {
+        ctx.device.push_error_scope(wgpu::ErrorFilter::Validation);
+
+        let _ = run_test(&ctx, &[0, 0, 0], false, true).await;
+
+        let error = pollster::block_on(ctx.device.pop_error_scope());
+        assert!(error.map_or(false, |error| {
+            format!("{error}").contains(
+                "Indirect buffer uses bytes 0..12 which overruns indirect buffer of size 0",
+            )
+        }));
+    });
+
 async fn run_test(
     ctx: &TestingContext,
     num_workgroups: &[u32; 3],
     forget_to_set_bind_group: bool,
+    set_buffer_size_to_0: bool,
```

Comment:
nitpick: I don't see coverage where both `forget_to_set_bind_group` and `set_buffer_size_to_0` are `true`, but that sounds interesting to do. Maybe we could add this to the `RESET_BIND_GROUPS` test?

Reply:
They each test validation individually though. A test where both would be true will have the same result as just `forget_to_set_bind_group` being true.

Reply:
Maybe this is indicative that we should use a 3-state `enum` instead of two `bool`s, then? 🤔

Reply:
This is no longer an issue with the refactored test file.